pkg/chains/storage/oci/oci.go

```diff
@@ -194,10 +198,88 @@ func (b *Backend) Type() string {
 	return StorageBackendOCI
 }
 
-func (b *Backend) RetrieveSignature(opts config.StorageOpts) (string, error) {
-	return "", fmt.Errorf("not implemented")
+func (b *Backend) RetrieveSignatures(opts config.StorageOpts) (map[string][]string, error) {
+	images, err := b.RetrieveArtifact(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string][]string)
+	for ref, img := range images {
+		sigImage, err := img.Signatures()
+		if err != nil {
+			return nil, err
+		}
+		sigs, err := sigImage.Get()
+		if err != nil {
+			return nil, err
+		}
+
+		signatures := []string{}
+		for _, s := range sigs {
+			if sig, err := s.Base64Signature(); err == nil {
+				signatures = append(signatures, sig)
+			}
+		}
+		m[ref] = signatures
+	}
+	return m, nil
+}
+
+func (b *Backend) RetrievePayloads(opts config.StorageOpts) (map[string]string, error) {
+	images, err := b.RetrieveArtifact(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]string)
+	for ref, img := range images {
+		attImage, err := img.Attestations()
```

Comment:
The `payload` here should depend on the value of `opts.PayloadFormat` correct? I believe if the `PayloadFormat` is `in-toto` then we want to call `img.Attestations()` here, but if the `PayloadFormat` is `simplesigning` then we'll want to call `img.Signatures()` here.

Reply:
Make sense 👍 Addressed ✅